paasta_tools/vitesscluster_tools.py

```diff
@@ -72,21 +662,98 @@ def validate(
             return []
 
 
-def load_vitess_instance_config(
+def generate_vitess_instance_config(
+    instance_config: Dict,
+) -> Dict:
+    # Generate the vitess instance config from the yelpsoa config
+
+    cpus = instance_config.get("cpus")
+    mem = instance_config.get("mem")
+    deploy_group = instance_config.get("deploy_group")
+    zk_address = instance_config.get("zk_address")
+    paasta_pool = instance_config.get("paasta_pool")
+    paasta_cluster = instance_config.get("paasta_cluster")
+    cells = instance_config.get("cells")
+    keyspaces = instance_config.get("keyspaces")
+    region = instance_config.get("region")
+
+    vitess_instance_config = {
+        "namespace": "paasta-vitessclusters",
+        "cpus": cpus,
+        "mem": mem,
+        "min_instances": 1,
+        "max_instances": 1,
+        "deploy_group": deploy_group,
+        "autoscaling": {"setpoint": 0.7},
+        "env": {
+            "OPERATOR_NAME": "vitess-operator",
+            "POD_NAME": "vitess-k8s",
+            "PS_OPERATOR_POD_NAME": "vitess-k8s",
+            "PS_OPERATOR_POD_NAMESPACE": "paasta-vitessclusters",
+            "WATCH_NAMESPACE": "paasta-vitessclusters",
+        },
+        "healthcheck_grace_period_seconds": 60,
+        "healthcheck_mode": "cmd",
+        "healthcheck_cmd": "true",
+        "images": {
+            "vtctld": VTCTLD_IMAGE,
+            "vtadmin": VT_ADMIN_IMAGE,
+            "vtgate": VT_GATE_IMAGE,
+            "vttablet": VT_TABLET_IMAGE,
+        },
+        "globalLockserver": {
+            "external": {
+                "implementation": TOPO_IMPLEMENTATION,
+                "address": zk_address,
+                "rootPath": TOPO_GLOBAL_ROOT,
+            }
+        },
+        "cells": [
+            build_cell_config(cell, paasta_pool, paasta_cluster, region)
+            for cell in cells
+        ],
+        "vitessDashboard": build_vitess_dashboard_config(
+            cells, paasta_pool, paasta_cluster, zk_address
+        ),
+        "vtadmin": build_vt_admin_config(cells, paasta_pool, paasta_cluster),
+        "keyspaces": build_keyspaces_config(
+            cells, keyspaces, paasta_pool, paasta_cluster, zk_address, region
+        ),
+        "updateStrategy": {"type": "Immediate"},
+    }
+    return vitess_instance_config
+
+
+def load_vitess_service_instance_configs(
     service: str,
     instance: str,
+    instance_type: str,
     cluster: str,
-    load_deployments: bool = True,
     soa_dir: str = DEFAULT_SOA_DIR,
-) -> VitessDeploymentConfig:
+) -> VitessDeploymentConfigDict:
     general_config = service_configuration_lib.read_service_configuration(
         service, soa_dir=soa_dir
     )
     instance_config = load_service_instance_config(
-        service, instance, "vitesscluster", cluster, soa_dir=soa_dir
+        service, instance, instance_type, cluster, soa_dir=soa_dir
     )
+    vitess_instance_config = generate_vitess_instance_config(Dict(instance_config))
```

Comment:
TIL that the typing.X classes can be used as normal constructors - that said, i don't think we need this if we update the parameter typing for this function to be `VitessDeploymentConfig`

Reply:
True, I used this to get over the failing mypy tox target, but the ideal fix would be to return of type VitessDeploymentConfigDict